release/mk-release-series.sh

```diff
@@ -49,9 +50,9 @@ fi
     git add doc/release-notes/release-${NEWVER}.sgml.in
 
   if `git diff HEAD` ; then
-    git commit -m "Branch ${NEWVER}.0.0"
+    git commit -m "${NEWVER}.0.0"
     git push
-    # github PR created by maintenance automation
+    gh pr create --repo squid-cache/squid --base master --fill
```

Comment:
Again, please detect/handle errors.

Reply:
I am aware that there are several types of error which could occur. But none of those can be automatically handled - they are all things like missing credentials, github permissions, or network outages. These do get reported to stderr so we can fix the cause and manually do the PR as workaround.

IMO the correct handling (such as can be automated) is already done by the `gh` tool internally. This script should ignore the lack of PR and continue with the process.

Reply:
> This script should ignore the lack of PR and continue with the process.

I strongly disagree. The default should be to quit and let the user handle the problem. If the user needs the process to continue, they can adjust the script accordingly.

Reply:
FWIW, I do not understand why you appear to be advocating that release maintainers perform secret changes to the release scripts.

This PR is a demonstration of what happens when "the user" attempts to "adjust the script accordingly" through the proper channels. Arguments with QA reviewer.

Reply:
> I do not understand why you appear to be advocating that release maintainers perform secret changes to the release scripts.

I do not advocate making releases (i.e. posting official PRs/etc.) with secret scripts. I am OK with folks adjusting any scripts while working on those scripts and/or on future releases (but not actually posting any official PRs/etc.), especially if such temporary/unofficial adjustments help improve the quality of the official script that actually makes the release at the end of that process.

> This PR is a demonstration of what happens when "the user" attempts to "adjust the script accordingly" through the proper channels. Arguments with QA reviewer.

This PR is a demonstration of what happens when new code that needs improvements is posted for review.

Reply:
In fact, a PR created by an official script that ignores errors should be treated similar to a PR created by a secret script (because the reviewer has no idea which errors the script has ignored before posting the PR -- the PR is effectively a result of an unknown code path execution)! My overall approach covers the two unwanted cases (i.e. error-ignoring official scripts creating official PRs and secret scripts creating official PRs) the same way (because their effect is similar): Avoid!

Compare those problems to a situation where a developer ran the official script that properly handles errors, that script failed, the developer made lots of "secret" (or not!) adjustments to properly rectify the situation, then rerun the official script that properly handles errors, and that script has finally succeeded, posting an official PR. In that situation, the developer still has a lot of freedom to properly address problems, but the resulting PR is indeed a product of a known official script -- a very desirable outcome!

Reply:
Let's not blow this out of proportion. There are several checks that happen during the workflow, both automated and manual, including on the correctness of the emitted PR.

We might argue that checks that are manual and later in the process are wastefui, but I am highly confident about the correctness of the outcome. PRs come and go, and even if one were to be created that needs to be immediately closed because it's incorrect is not the end of the world. 
This said,
```
set -eu -o pipefail
```
at the beginning of the script will make all these problems go away, because it will make it so that any unhandled error anywhere in a pipeline will cause the script to exit with return code as the rightmost failing command in the pipeline

Reply:
I hope that "Let's not blow this out of proportion" remark was in response to an earlier accusation that I am "advocating that release maintainers perform secret changes to the release scripts." rather than my basic request to detect and handle errors or my explanation that I am not guilty of the alleged advocacy.

There is nothing I can do to counter the "I am highly confident about the correctness of the outcome" and "not the end of the world" arguments in response to a basic request to detect and handle errors. I should probably just stop wasting time on reviewing this code.